### DISCUSSION TITLE
Permissions required

### DISCUSSION THREAD

**Discussion by Rcarr1** [pnp/powershell#4418] [General]
We are using PnP.Powershell to place a document on Sharepoint every 10 minutes via automation and service account. We've registered the app with Azure per instructions. We granted the service account permissions (both Admin and owner) on the site. We've granted the App Sites.Selected to Sharepoint and gave it AllSites.FullControl briefly while we attempted to grant permissions to the site we need via this command: "grant-PnPAzureADAppSitePermission -AppId "ApplicationID" -DisplayName "PnP Management Shell" -Permissions Write -Site "https://tenant.sharepoint.com/sites/site""

This was unsuccessful and resulted in an error message stating we need further permissions from the Global Admin. What specific permissions do we need to grant to get this to function? We're using v2.12.0. We fully intend to revoke the AllSites.FullControl for security reasons and want to grant it the minimal permissions needed to function on this site only. 

Can someone please advise on what permissions are needed for this?

**Comment by Barbarur** [comment]
What is the full error?
AllSites.FullControl is given for Graph or SPO API?
Are you using Application or Delegated permissions?
